### PR TITLE
Expose ims data for sending personal messages.

### PR DESCRIPTION
**Pull request by mgwidmann**

### CODE REVIEW

**Comment by mgwidmann** [comment]
@BlakeWilliams ping

**Comment by BlakeWilliams** [comment]
Sorry for the delay!

I think this looks good but we also need to collect IM's as they're added.

A good place to add that would be here: https://github.com/BlakeWilliams/Elixir-Slack/blob/master/lib/slack/handlers.ex

I haven't looked into it but I don't think has a specific "IM received" event we can hook into.

**Comment by mgwidmann** [comment]
Not really seeing an event like that:

https://api.slack.com/rtm

I'm afraid we'll have to poll https://api.slack.com/methods/im.list which may be undesirable.

**Comment by BlakeWilliams** [comment]
I don't think that would be the only solution. We might have to have this watch messages and check if it matches a user instead of a channel.

**Comment by mgwidmann** [comment]
So it looks like we are listening for user changes here:
https://github.com/BlakeWilliams/Elixir-Slack/blob/master/lib/slack/handlers.ex#L39
Waiting for an answer here: https://twitter.com/SlackAPI/status/665241686633816064 but it seems that `im.list` data is not returned. Heres an example of an IM vs a User:

```elixir
# Example IM with data changed for privacy
%{"D0B51234" => %{created: 1442969439, has_pins: false, id: "D0B51234",
    is_im: true, is_open: true, last_read: "0000000000.000000", latest: nil,
    unread_count: 0, unread_count_display: 0, user: "U07HP1234"},
}
# Example User with data changed for privacy
%{"U03CC1234" => %{color: "619a4f", deleted: false, id: "U03CC1234",
    is_admin: true, is_bot: false, is_owner: false, is_primary_owner: false,
    is_restricted: false, is_ultra_restricted: false, name: "username",
    presence: "away",
    profile: %{email: "username@example.com", fields: nil,
      first_name: "User",
      image_192: "...",
      image_24: "...",
      image_32: "...",
      image_48: "...",
      image_512: "...",
      image_72: "...",
      last_name: "Name", real_name: "User Name",
      real_name_normalized: "User Name",
      title: "User User Name"}, real_name: "User Name",
    status: nil, team_id: "T02UF1234", tz: "America/Indiana/Indianapolis",
    tz_label: "Eastern Standard Time", tz_offset: -18000}
}
```
The data does not align at all.

@BlakeWilliams what do you want to do?

**Comment by BlakeWilliams** [comment]
What's wrong with checking `is_im` and adding it to the im list?

File: lib/slack/handlers.ex [hunk 1]
```diff
@@ -48,6 +48,14 @@ defmodule Slack.Handlers do
     end
   end)
 
+  def handle_slack(%{type: "im_created", channel: channel, user: user}, slack) do
+    {:ok, put_in(slack, [:ims, channel.id], %{id: channel.id, user: user})}
+  end
+
+  def handle_slack(%{type: "im_close", channel: channel}, slack) do
+    {:ok, put_in(slack, [:ims], Map.delete(slack[:ims], channel))}
+  end
```
Comment (mgwidmann) [inline lib/slack/handlers.ex]: @BlakeWilliams Added im create/close events here, these are fired after the [im.open](https://api.slack.com/methods/im.open) and [im.close](https://api.slack.com/methods/im.close) methods are called.

To make it easier on the user, I've added a `Slack.Web` module which has the beginnings of the Web API for areas like this where websocket support is lacking.

**Comment by mgwidmann** [comment]
See outdated diff: https://github.com/BlakeWilliams/Elixir-Slack/pull/26#discussion_r46512747

**Comment by mgwidmann** [comment]
@BlakeWilliams ping

**Comment by BlakeWilliams** [comment]
I think this will be resolved by #55 as well. If anything is missing feel free to open an issue.

**Pull request closed without merging**